### PR TITLE
[Bug] Fix `release-*` Github Workflows and Review `integrations-pr` command

### PR DESCRIPTION
**Pull request by terrancedejesus**
## Issues
* https://github.com/elastic/detection-rules/issues/2335
* https://github.com/elastic/detection-rules/issues/2244

## Summary
The following Github workflows fail to run correctly when attempting to clone necessary repositories.

`release-fleet`
`release-kibana`

We may consider removing the `release-kibana` workflow as releasing to Kibana is no longer necessary.

## Release Fleet Bugs
There may be additional bugs in this that have not been found yet as a result of the workflow exiting at each failed step.

<details>
<summary> 🐛 1: Unable to checkout `elastic/integrations` repository </summary>

Screenshot 2023-02-27 at 10 21 21 AM

Potential reasons explored:
- ~Order of operations in workflow~
- ~Permissions with existing token~ (Tried PAT with excessive permissions)
- ~`elastic/` remove from ref~ (Complained about missing org)
- ~Since target repo is public, remove token usage~
- ~Update to checkout@v3~
- ~Tried R/W token~ (New error: `exit code 128` or user not found, potentially expired)

*Solution:* Removed `ref: ${{github.event.inputs.target_branch}}` which points to master

</details>

<details>
<summary> 🐛 2: Unable to install `elastic-package` Go binary </summary>

Screenshot 2023-02-27 at 11 50 06 AM

Potential reasons explored:

- ~Change `go get` to `go install @latest`~

*Solution:* Change `go get` to `go install @latest`

</details>

<details>
<summary>🐛 3: `integrations-pr` CLI command fails targeting repo</summary>

Screenshot 2023-02-27 at 12 17 25 PM

Potential reasons explored:

- ~Changed the installation location of integrations~
- ~Use the latest Go setup~
- ~Since master is not used to get integrations, remove from CLI command~

*Solution:* Change target_branch to `main`

</details>

<details>
<summary>🐛 4: `integrations-pr` CLI command fails to create PR based on permissions </summary>

Screenshot 2023-02-27 at 12 46 01 PM

Potential reasons explored:

- ~PAT does not have correct permissions to read and write to integrations repo~

*Solution:* Created a new token that had the correct permissions to create pull requests to the target repository

</details>

## Release Fleet Enhancements

<details>
<summary> ➕ Add step to bump package versions, dynamically determine next ga or beta version</summary>
This enhancement will dynamically determine the next ga or beta version, then save it to the `packages.yml` file prior to the build package release step.

Screenshot 2023-02-28 at 11 15 18 AM
Screenshot 2023-02-28 at 11 29 18 AM
Screenshot 2023-02-28 at 11 41 42 AM

</details>

## `integrations-pr` Command Bugs
<details>
<summary>🐛 1: `integrations-pr` CLI command `--patch-minor` </summary>
If the last package version was `8.7.2.beta.1` and we are on a branch that never had a package, the next package will be `8.7.3.beta.1`, not `8.8.0.beta.1`

Screenshot 2023-02-28 at 11 15 18 AM
Screenshot 2023-02-28 at 11 41 42 AM

</details>

## Testing
The following can be used for peer review and testing.

1. Navigate to Actions
2. Select `release-fleet` workflow
3. Select `run-workflow`
4. Change `Use workflow from: main` to `Use workflow from: 2335-bug-fix-the-two-release-workflows`

⚠️ Since the PR is to main, it will use the `packages.yml` contents where the package version will be `8.8-beta.1`. This will also be used to create the integrations branch locally and create the PR. Therefore, if multiple people are testing, there may be conflicting names.

To solve this:
1. Go to `https://github.com/elastic/integrations/pulls` and find the corresponding PR
2. Close with `This is testing only by TRaDE regarding pre-built rules package automation`
3. Go to `https://github.com/elastic/integrations/branches` and delete the corresponding branch

### Expected Results

✅ Workflow completed successfully | [Example](https://github.com/elastic/detection-rules/actions/runs/4285497835)
✅ Integrations PR created | [Example](https://github.com/elastic/integrations/pull/5413)

### CODE REVIEW

File: .github/workflows/release-fleet.yml [hunk 1]
```diff
@@ -28,15 +28,14 @@ jobs:
...
+        uses: actions/checkout@v3
         with:
           path: detection-rules
 
       - name: Checkout elastic/integrations
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3
         with:
-          token: ${{ secrets.PROTECTIONS_MACHINE_TOKEN }}
-          ref: ${{github.event.inputs.target_branch}}
```
Comment (terrancedejesus) [inline .github/workflows/release-fleet.yml:39]: Removing the reference to the master seemed to help fetch the integrations repo successfully.

Comment (brokensound77) [reply .github/workflows/release-fleet.yml:39]: so strange - I wish we understood why

File: .github/workflows/release-fleet.yml [hunk 2]
```diff
@@ -51,6 +53,15 @@ jobs:
           python -m pip install --upgrade pip
           pip install .[dev]
 
+      - name: Bump prebuilt rules package version
+        env:
+          PACKAGE_MATURITY: "${{github.event.inputs.package_maturity}}"
```
Comment (brokensound77) [inline .github/workflows/release-fleet.yml:58]: how do we handle _major_ or non standard bumps? Could expose it as an unrequired input, where if undefined, it uses this

**Review by brokensound77 — APPROVED** [review]
LGTM, nice job!

**Review by eric-forte-elastic — APPROVED** [review]
LGTM, great peer review session yesterday and manual code review this morning.

File: .github/workflows/release-fleet.yml (hunk 2 above)
Comment (terrancedejesus) [reply .github/workflows/release-fleet.yml:58]: We will run the workflow from a specific branch where `packages.yml` is different at each branch but essentially the stack versions match that branch.

Therefore if we run from 9.0 branch, the registry version will already be `9.0.0.beta.1` and kibana stack version is `^9.0.0`, therefore we are only concerned about the beta and ga package versions for the patch version.

Workflow wise:

branch: `9.0`
package version default: `9.0.0-beta.0`
determined release fleet package version `9.0.0-beta.1`

From `integrations.py` we use `find_latest_integration_version()` which would return, for example, `8.10.1-beta.2` as the latest version in EPR.

In [L61](https://github.com/elastic/detection-rules/blob/315e3cf1b7215168cf0e5955cc6a7999366cfbee/.github/workflows/release-fleet.yml#L61) we bump package versions based on the latest (`8.10.1-beta.2`) and maturity `beta`. In the command the following should catch such examples for a major or minor that does not have an existing package.

```
# if an existing minor or major does not have a package, bump from the last
# example is 8.10.0-beta.1 is last, but on 9.0.0 major
# example is 8.10.0-beta.1 is last, but on 8.11.0 minor
if latest_patch_release_ver.minor != pkg_kibana_ver.minor:
    latest_patch_release_ver = latest_patch_release_ver.bump_minor()
if latest_patch_release_ver.major != pkg_kibana_ver.major:
    latest_patch_release_ver = latest_patch_release_ver.bump_major()
```

**Pull request merged**